kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -99,28 +115,90 @@ def before_command_run(
             )
 
 
-def _format_user_cli_data(
-    hashed_username: str, command_args: List[str], project_metadata: ProjectMetadata
+class KedroTelemetryHooks:  # pylint: disable=too-few-public-methods
+    """Hook to send proejct statistics data to Heap"""
+
+    project_properties = PROJECT_PROPERTIES if PROJECT_PROPERTIES else {}
+
+    @hook_impl
+    def after_context_created(self, context):
+        """Hook implementation to send proejct statistics data to Heap"""
+
+        catalog = context.catalog
+        default_pipeline = pipelines.get("__default__")
+        hashed_username = _get_hashed_username()
+
+        if not self.project_properties:  # `KedroSession.run` without invoking CLI
+            project_metadata = bootstrap_project(context.project_path)
+            _prepare_project_properties(
+                self.project_properties, hashed_username, project_metadata
+            )
+
+        project_statistics_properties = _format_project_statistics_data(
+            self.project_properties, catalog, default_pipeline, pipelines
+        )
+
+        _send_heap_event(
+            event_name="Kedro Project Statistics",
+            identity=hashed_username,
+            properties=project_statistics_properties,
+        )
+
+
+def _prepare_project_properties(
+    properties: dict, hashed_username: str, project_metadata: ProjectMetadata
```

Comment:
I think it would be useful to add a docstring about what this method does, since it doesn't return anything.

Reply:
Obsoleted as the `PROJECT_PROPERTIES` is killed now and this function now return a new dict instead.